### ISSUE TITLE
No Audio at all in PLAYERUNKNOWN'S BATTLEGROUNDS

### ISSUE THREAD

**Issue by ToxMox** [eiz/SynchronousAudioRouter#27]
In PUBG I get no audio when using SAR. If I reboot and don't start Reaper then I have audio in the game. I'm not even trying to route that game's audio via SAR. BattleEye blocks the sar dll but it always did that and It used to work so I'm guessing they changed something. Any insight?

To be clear even if I close Reaper and the SAR virtual devices go away I still I get no audio in the game until I restart the PC. Once I run Reaper once and SAR activates then no more audio.

**Comment by pannal** [comment]
I've already created a ticket with Battleye on this issue.

**Comment by pannal** [comment]
I've gotten an answer from EN MASSE (subsidiary of Bluehole, seems to handle PUBG support):
```
I appreciate your report, and have forwarded it to the appropriate team for investigation. We are already aware of this issue, but your report will help us to further our investigation, and hopefully fix this as soon as possible. Thank you!
```

**Comment by jscinoz** [comment]
Unsurprisingly, Fortnite has the same issue as it also uses Battleye. Is there a public link to the Battleye ticket, or if not would there happen to have been a response yet?

**Comment by pannal** [comment]
No response other than that they know about the issue and "redirected it to the proper people".

**Comment by jerryen** [comment]
0.12-beta works perfectly fine with BattlEye so if you don’t need the 0.13 fixes, installing 0.12 is probably the way to go for now.

**Comment by pannal** [comment]
Interesting, @eiz might this be related to the NDIS driver which was added with 0.13? Could we get a test build of 0.13 without that driver?

**Comment by pannal** [comment]
This is invalid. 0.13 works perfectly with PUBG/BattlEye and REAPER. Just don't run REAPER elevated.

**Comment by AKPWebDesign** [comment]
@pannal Are you sure about this? I'm using 0.13 and I'm unable to play PUBG or Fortnite after loading SAR (and that isn't solved until I restart the computer, simply unloading SAR doesn't work). I don't run REAPER elevated.

**Comment by pannal** [comment]
Yes I am, I've played several hours yesterday using SAR. I think it's installed to not require admin rights and reaper doesn't run elevated.

**Comment by AKPWebDesign** [comment]
Interesting. I've got the exact same setup, SAR installed allowing non-admin use and REAPER running non-elevated. Are you running 64-bit or 32-bit REAPER and SAR?

**Comment by pannal** [comment]
Both x64. I have the BEService on CPU priority idle and pinned to one core using ProcessLasso, but I don't think that's what makes the key difference.

**Comment by AKPWebDesign** [comment]
Yeah, I doubt that would make a difference, but I might try it anyway. I've also got both running x64 and I'm seeing SarASIO_x64.dll being blocked by BattleEye.

**Comment by pannal** [comment]
I'm not sure what else could be different.

**Comment by ToxMox** [comment]
I'm assuming you rebooted? I haven't even tested yet unelevated but when I did have the issue originally I needed to reboot before audio worked for the game again.

**Comment by pannal** [comment]
And make sure you're actually on 0.13. Even after closing REAPER, if something (especially Skype) holds a reference to the audio device, the (repair) install of SAR wouldn't go through properly for me. It went through but didn't actually replace the files.

**Comment by NightmareJoker2** [comment]
@AKPWebDesign What kind of hardware sound devices are you using?
What kind of ASIO drivers, other than SAR, do you have installed?
If you don't have any hardware specific ASIO drivers, is your [ASIO4ALL](http://www.asio4all.org/) driver up to date? (sha256 for 2.14 is `15a3a1ff103f583f9e5493349c85b98799baadee44cb3f25a0611141a4a6d8fa`)
Are all ASIO applications truly closed when sound output fails?
Please be aware that the audio device you select for output through ASIO will be exclusively locked to the application using it by the ASIO driver, preventing you from hearing other applications through it via the default Windows Sound API. You will need to set up your capture rules such that output is provided, but selectively duplicated to another interface which you route to where you need it, or capture through a Windows Sound API compatible application.
BattlEye does not block it, and if it did, it would not tell you about it unless you are working with the BattlEye developer kit, it would simply not work or cause the protected application to quit (and optionally log it on the game server). The source of your problem is likely elsewhere.

**Comment by AKPWebDesign** [comment]
@NightmareJoker2 I've got a Focusrite Scarlett Solo and its associated ASIO driver.

>BattlEye does not block it, and if it did, it would not tell you about it

I hate to be that guy, but it seems to be pretty clear to me that BattlEye is blocking the DLL... https://i.imgur.com/xwSVLov.png

**Comment by NightmareJoker2** [comment]
@AKPWebDesign That's odd.
The launcher you have provided a screenshot of is also not from **PLAYERUNKNOWN's BATTLEGROUNDS**.

1. Try updating or reinstalling your game.
2. Make sure the file in that location isn't corrupted or out of date. `SarAsio_x64.dll` needs to be `855552 bytes` in size and have a SHA-256 checksum of `edaebef5abea4328d78fd780812b34324b017636f710373daed50fcd3c3bf294` or have a valid digital signature to be loaded.
3. All applications need to run in an unelevated application context (not run as Administrator) for it to work properly.

**Comment by AKPWebDesign** [comment]
@NightmareJoker2 You're absolutely right, that's Fortnite. That being said, PUBG did the same thing before the 1.0 release. I uninstalled the game due to space constraints, but just reinstalled and verified that my setup is working with PUBG now. I'm doing everything the same way as I was before though, so I can only assume it wasn't a change that I made. Either way, PUBG works now, so that's cool. If only it worked with Fortnite too...

**Comment by pannal** [comment]
@AKPWebDesign Cool, that indicates they listened to the bug reports. I've made contact with them on several occasions - perhaps their BattlEye version is newer now compared to Fortnite?

**Comment by zharf** [comment]
yeah I'm currently having the same issue with arma 3...

**Comment by pannal** [comment]
This is happening again.

**Comment by zharf** [comment]
unfortunately I've not received any response from battleye on the issue :(

**Comment by pannal** [comment]
I've wrote to both of them the day before yesterday (again) - since yesterday evening it works again. Someone listened.
